### DISCUSSION TITLE
Native iOS dApp Browser

### DISCUSSION THREAD

**Discussion by jessejustx** [WalletConnect/walletconnect-monorepo#2452] [Developer Support]
Hello team, first of all thank you for providing the community with your service our team high appreciates your work!
I would like to ask some questions if that would be ok.
1. Our team is creating a wallet and in the wallet we intend to add an in app dApp browser.
2. This browser aims to work similarly to coinbase wallet / zerion etc.
3. By this I mean automatically reconnecting to dApp websites like opensea when accessed without any user triggers.

Could I have some help or guidelines on how to implement this behavior?
Help would very much be appreciated :(
Thank you!

**Comment by finessevanes** [comment]
Hi there! Thank you for the kind words and we're happy to help. To clarify, are you looking for guidance on how to automatically reconnect to dapp websites without any user triggers within your app's browser?

**Reply by jessejustx** [reply]
Thank you for replying!
Yes and also I want the dApp (such as opensea) to automatically prompt a connection when the url is loaded in my webview.
I have seen other apps implement this behavior but cant seem to get my head around it :(

**Reply by finessevanes** [reply]
You'll probably want to implement a store to track the user's session with the dapp. When the user comes back to the dapp, if a session exists and automatically reconnect them to the dapp without any user intervention.

**Comment by ablankstory** [comment]
@jessejustx - did you have any luck here? especially around the automatically prompt a connection when the url is loaded into the wallet's dapp browser?

right now, we're having an issue when users select "Connect > WalletConnect > [our wallet]". The connection just spins and doesn't prompt the user to sign.
